functions/src/updateUserRecommendations.js

```diff
@@ -14,37 +14,54 @@ const ACTIVITY_TYPES = {
 };
 
 exports.handler = userKey =>
-  admin
-    .database()
-    .ref("/paths")
-    .orderByChild("isPublic")
-    .equalTo(true)
-    .once("value")
-    .then(snapshot => snapshot.val())
-    .then(paths =>
-      Promise.all(
-        Object.keys(paths).map(pathKey =>
-          Promise.all([
-            admin
-              .database()
-              .ref("/activities")
-              .orderByChild("path")
-              .equalTo(pathKey)
-              .once("value")
-              .then(snapshot => snapshot.val()),
-            admin
-              .database()
-              .ref(`/completedActivities/${userKey}/${pathKey}`)
-              .once("value")
-              .then(snapshot => snapshot.val())
-          ])
+  Promise.all([
+    admin
+      .database()
+      .ref("/activityExampleSolutions")
+      .once("value")
+      .then(snapshot => snapshot.val()),
+    admin
+      .database()
+      .ref(`/userCodingSkills/${userKey}`)
+      .once("value")
+      .then(snapshot => snapshot.val()),
+    admin
+      .database()
+      .ref("/paths")
+      .orderByChild("isPublic")
+      .equalTo(true)
+      .once("value")
+      .then(snapshot => snapshot.val())
+      .then(paths =>
+        Promise.all(
+          Object.keys(paths).map(pathKey =>
+            Promise.all([
+              admin
+                .database()
+                .ref("/activities")
+                .orderByChild("path")
+                .equalTo(pathKey)
+                .once("value")
+                .then(snapshot => snapshot.val()),
+              admin
+                .database()
+                .ref(`/completedActivities/${userKey}/${pathKey}`)
+                .once("value")
+                .then(snapshot => snapshot.val())
+            ])
+          )
         )
       )
-    )
-    .then(pathActivitiesData => {
+  ])
+    .then(([activityExampleSolutions = {}, userSkills = {}, pathActivitiesData]) => {
       const result = {};
       const updated = {};
-
+      const problemWithUnsolvedSkills = {
+        title: "Jupyter Activities With New Skills"
+      };
+      const problemWithSolvedSkills = {
```

Comment:
@scboesch is this to recommend users activities to reinforce their existing knowledge?